recipes/qt/6.x.x/conanfile.py

```diff
@@ -488,33 +489,73 @@ def build(self):
                     with tools.run_environment(self):
                         cmake.build()
+    @property
+    def _cmake_executables_file(self):
+        return os.path.join("lib", "cmake", "Qt6Core", "conan_qt_executables_variables.cmake")
 
     def package(self):
         cmake = self._configure_cmake()
         cmake.install()
         with open(os.path.join(self.package_folder, "bin", "qt.conf"), "w") as f:
-            f.write("""[Paths]
-Prefix = ..
-ArchData = res/archdatadir
-HostData = res/archdatadir
-Data = res/datadir
-Sysconf = res/sysconfdir
-LibraryExecutables = res/archdatadir/bin
-Plugins = res/archdatadir/plugins
-Imports = res/archdatadir/imports
-Qml2Imports = res/archdatadir/qml
-Translations = res/datadir/translations
-Documentation = res/datadir/doc
-Examples = res/datadir/examples""")
+            f.write(textwrap.dedent("""[Paths]
+                Prefix = ..
+                ArchData = res/archdatadir
+                HostData = res/archdatadir
+                Data = res/datadir
+                Sysconf = res/sysconfdir
+                LibraryExecutables = res/archdatadir/bin
+                Plugins = res/archdatadir/plugins
+                Imports = res/archdatadir/imports
+                Qml2Imports = res/archdatadir/qml
+                Translations = res/datadir/translations
+                Documentation = res/datadir/doc
+                Examples = res/datadir/examples"""))
         self.copy("*LICENSE*", src="qt6/", dst="licenses")
         for module in self._submodules:
             if not self.options.get_safe(module):
                 tools.rmdir(os.path.join(self.package_folder, "licenses", module))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        for mask in ["Find*.cmake", "*Config.cmake", "*-config.cmake"]:
+            tools.remove_files_by_mask(self.package_folder, mask)
         tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la*")
         tools.remove_files_by_mask(self.package_folder, "*.pdb*")
         os.remove(os.path.join(self.package_folder, "bin", "qt-cmake-private-install.cmake"))
+        
+        for m in os.listdir(os.path.join(self.package_folder, "lib", "cmake")):
+            module = os.path.join(self.package_folder, "lib", "cmake", m, "%sMacros.cmake" % m)
+            if not os.path.isfile(module):
+                tools.rmdir(os.path.join(self.package_folder, "lib", "cmake", m))
+
+        extension = ""
+        if self.settings.os == "Windows":
+            extension = ".exe"
+        filecontents = "set(QT_CMAKE_EXPORT_NAMESPACE Qt6)\n"
+        filecontents += "set(QT_VERSION_MAJOR %d)\n" % int(self.version.split(".")[0])
+        filecontents += "set(QT_VERSION_MINOR %d)\n" % int(self.version.split(".")[1])
+        filecontents += "set(QT_VERSION_PATCH %d)\n" % int(self.version.split(".")[2])
```

Comment:
Would this be an improvement?

```suggestion
        ver = tools.Version(self.version)
        filecontents += "set(QT_VERSION_MAJOR %d)\n" % ver.major)
        filecontents += "set(QT_VERSION_MINOR %d)\n" % ver.minor)
        filecontents += "set(QT_VERSION_PATCH %d)\n" % ver.patch)
```

Reply:
For sure, thanks